applications/registration-api/src/main/java/no/dcat/controller/ApiRegistrationController.java

```diff
@@ -0,0 +1,288 @@
+
+    ApiRegistration apiRegistration =
+        apiRegistrationOptional.isPresent() ? apiRegistrationOptional.get() : null;
+
+    if (apiRegistration == null || !Objects.equals(orgNr, apiRegistration.getOrgNr())) {
+      return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    return new ResponseEntity<>(apiRegistration, HttpStatus.OK);
+  }
+
+  /**
+   * Return list of all apiRegistrations in catalog. Without parameters, the first 20
+   * apiRegistrations are returned The returned data contains paging hyperlinks.
+   *
+   * <p>
+   *
+   * @param orgNr the id of the catalog
+   * @param pageable number of apiRegistrations returned
+   * @return List of data sets, with hyperlinks to other pages in search result
+   */
+  @PreAuthorize("hasPermission(#orgNr, 'write')")
+  @CrossOrigin
+  @RequestMapping(value = "", method = GET, produces = APPLICATION_JSON_UTF8_VALUE)
+  public HttpEntity<PagedResources<ApiRegistration>> listApiRegistrations(
+      @PathVariable("orgNr") String orgNr,
+      Pageable pageable,
+      PagedResourcesAssembler assembler) {
+
+    logger.info("Get apiRegistrations");
+
+    Page<ApiRegistration> apiRegistrations =
+        apiRegistrationRepository.findByOrgNr(orgNr, pageable);
+    return new ResponseEntity<>(assembler.toResource(apiRegistrations), HttpStatus.OK);
+  }
+
+  /**
+   * Create new apiRegistration in catalog. ID for the apiRegistration is created automatically.
+   *
+   * @param apiRegistration
+   * @return HTTP 200 OK if apiRegistration could be could be created.
+   */
+  @PreAuthorize("hasPermission(#orgNr, 'write')")
+  @CrossOrigin
+  @RequestMapping(
+      value = "/",
+      method = POST,
+      consumes = APPLICATION_JSON_VALUE,
+      produces = APPLICATION_JSON_UTF8_VALUE)
+  public HttpEntity<ApiRegistration> saveApispec(
+      @PathVariable("orgNr") String orgNr, @RequestBody ApiRegistration apiRegistration)
+      throws CatalogNotFoundException {
+
+    logger.info("SAVE requestbody apiRegistration");
+
+    Optional<Catalog> catalogOptional = catalogRepository.findById(orgNr);
+
+    if (!catalogOptional.isPresent()) {
+      throw new CatalogNotFoundException(
+          String.format(
+              "Unable to create apiRegistration, catalog with id %s not found", orgNr));
+    }
+
+    ApiRegistration savedApispec =
+        createAndSaveApispec(orgNr, apiRegistration, catalogOptional.get());
+
+    return new ResponseEntity<>(savedApispec, HttpStatus.OK);
+  }
+
+  ApiRegistration createAndSaveApispec(
+      String orgNr, ApiRegistration apiRegistration, Catalog catalog) {
+
+    // Create new apiRegistration
+    ApiRegistration apiSpecWithNewId = RegistrationFactory.createApiRegistration(orgNr);
+
+    // force new id, uri and catalogId, to ensure saving
+    apiRegistration.setId(apiSpecWithNewId.getId());
+
+    apiRegistration.setOrgNr(apiSpecWithNewId.getOrgNr());
```

Comment:
Jeg kan ikke forstå noe nå. Hvorfor trenger vi to objekter og hvor kommer id fra.

Reply:
Dette kan forenkles mer, id er en randomUUID.